indico/modules/events/editing/controllers/backend/timeline.py

```diff
@@ -388,7 +388,11 @@ def _process(self):
                 folder_name = secure_filename(file_type.name, f'file-type-{file_type.id}')
 
                 with file.storage.get_local_path(file.storage_file_id) as filepath:
-                    zip_handler.write(filepath, os.path.join(folder_name, filename))
+                    info = ZipInfo.from_file(filepath)
+                    info.filename = os.path.join(folder_name, filename)
```

Comment:
Isn't this already already set in the ZipInfo constructor?

Reply:
The problem is I'm getting the ZipInfo object from the `from_file` util, which automatically grabs all info such as permissions, but then I have to set the filename we want for the file in the zip, otherwise it will have a path like `<INDICO DIR>/indico/data/archive/event/<EVENT>/editing/...` inside the zip...

Would using the ZipInfo constructor be a better option here?

Reply:
OK this makes perfect sense, there's in deed nothing else that sets the file name used inside the ZIP.

However, I now realize that passing `src.name` above probably won't always work. This may be AND kind of file object, which doesn't necessarily have a name or physical filesystem location (in case of a file stored on S3 for example).

Reply:
Hmm in this case it might be better to use the ZipInfo constructor then. I think there won't be any problems if we always set the permissions to be `644`, right?

Reply:
that should be fine